File: Lua/TestStack.lua
require("Stack")

-- TestStack Script
-- December 10, 2017
-- Demonstration of the Stack ADT in Lua.

local stack = STACK:Create()

-- Ask for user input until end condition is met.
while true do
	io.write("Please enter a non-negative number to convert to binary: \n")
	input = io.read("*n")

	-- If negative is entered, break loop.
	if input < 0 then break end

	-- User our "Stack" for binary conversion.
	repeat
		stack:push(input % 2)
		input = math.floor(input / 2)
	until input == 0

	-- Pop elements of out "Stack" to print conversion.
	while stack:top() do
		io.write(stack:pop())
	end

	-- Prepare new lines for next input.
	io.write("\n\n")

end

File: Java/TestStack.java
import java.util.Scanner;

/*
TestStack Class
December 10, 2017
Demonstration of the Stack ADT in Java.
*/
public final class TestStack {
    
    public static void main(String[] args) {
        Stack<Integer> stack = new Stack();
        Scanner scanner = new Scanner(System.in);
        
        // Ask for user input until end condition is met.
        for(;;) {
            System.out.println("Please enter a non-negative number to convert to binary: ");
            int input = scanner.nextInt();
            
            // If negative is entered, break loop.
            if(input < 0) break;
            
            // Use Stack<T> for binary conversion.
            do {
                stack.push(input % 2);
                input /= 2;
            } while(input != 0);
            
            // Pop elements of Stack<T> to print conversion.
            while(stack.top() != null)
                System.out.print(stack.pop());
            
            // Prepare new lines for next input.
            System.out.println("\n");
        }
    }
}
